cni/pkg/install-cni/pkg/install/kubeconfig.go

```diff
@@ -108,40 +110,23 @@ func createKubeconfigFile(cfg *config.Config, saToken string) (kubeconfigFilepat
 		TLSConfig:                 tlsConfig,
 	}
 
-	var tmpFile *os.File
-	tmpFile, err = ioutil.TempFile(cfg.MountedCNINetDir, cfg.KubeconfigFilename+".tmp.")
-	if err != nil {
-		return
-	}
-	defer func() {
-		if file.Exists(tmpFile.Name()) {
-			if rmErr := os.Remove(tmpFile.Name()); rmErr != nil {
-				if err != nil {
-					err = errors.Wrap(err, rmErr.Error())
-				} else {
-					err = rmErr
-				}
-			}
-		}
-	}()
-
-	if err = os.Chmod(tmpFile.Name(), os.FileMode(cfg.KubeconfigMode)); err != nil {
-		return
+	var kcbb bytes.Buffer
+	if err := tpl.Execute(&kcbb, fields); err != nil {
+		return "", err
 	}
 
-	if err = tpl.Execute(tmpFile, fields); err != nil {
-		if closeErr := tmpFile.Close(); closeErr != nil {
-			err = errors.Wrap(err, closeErr.Error())
-		}
-		return
+	var kcbbToPrint bytes.Buffer
+	fields.ServiceAccountToken = "<redacted>"
+	if !cfg.SkipTLSVerify {
+		fields.TLSConfig = fmt.Sprintf("certificate-authority-data: <CA cert from %s>", caFile)
 	}
-
-	if err = tmpFile.Close(); err != nil {
-		return
+	if err := tpl.Execute(&kcbbToPrint, fields); err != nil {
+		return "", err
 	}
 
 	kubeconfigFilepath = filepath.Join(cfg.MountedCNINetDir, cfg.KubeconfigFilename)
-	if err = os.Rename(tmpFile.Name(), kubeconfigFilepath); err != nil {
+	log.Infof("write kubeconfig file %s with: \n%+v", kubeconfigFilepath, kcbbToPrint.String())
+	if err = os.WriteFile(kubeconfigFilepath, kcbb.Bytes(), os.FileMode(cfg.KubeconfigMode)); err != nil {
```

Comment:
I remember a lot about making writes in cni atomic - do we need to use file.AtomicWrite? It seems like the old way was?

Reply:
Done.